TankClient.java

```diff
@@ -91,442 +109,424 @@ public void framPaint(Graphics graphics) {
+			tr.draw(graphics);
+		}
 
-			for (int i = 0; i < homeWall.size(); i++) {
-				CommonWall w = homeWall.get(i);
-				homeTank.collideWithWall(w);
-				if (player2)homeTank2.collideWithWall(w);
-				w.draw(graphics);
-			}
+		for (int i = 0; i < bombTanks.size(); i++) {
+			BombTank bt = bombTanks.get(i);
+			bt.draw(graphics);
+		}
 
+		for (int i = 0; i < otherWall.size(); i++) {
+			CommonWall cw = otherWall.get(i);
+			cw.draw(graphics);
 		}
 
-		public TankClient() {
-
-			jmb = new MenuBar();
-			jm1 = new Menu("Game");
-			jm2 = new Menu("Pause/Continue");
-			jm3 = new Menu("Help");
-			jm4 = new Menu("Level");
-			jm5 = new Menu("Addition");
-			jm1.setFont(new Font(TIMES_NEW_ROMAN, Font.BOLD, 15));
-			jm2.setFont(new Font(TIMES_NEW_ROMAN, Font.BOLD, 15));
-			jm3.setFont(new Font(TIMES_NEW_ROMAN, Font.BOLD, 15));
-			jm4.setFont(new Font(TIMES_NEW_ROMAN, Font.BOLD, 15));
-
-			jmi1 = new MenuItem("New Game");
-			jmi2 = new MenuItem("Exit");
-			jmi3 = new MenuItem("Stop");
-			jmi4 = new MenuItem("Continue");
-			jmi5 = new MenuItem("Help");
-			jmi6 = new MenuItem("Level1");
-			jmi7 = new MenuItem("Level2");
-			jmi8 = new MenuItem("Level3");
-			jmi9 = new MenuItem("Level4");
-			jmi10=new MenuItem("Add Player 2");
-			jmi11= new MenuItem("Join other's game");
-			jmi1.setFont(new Font(TIMES_NEW_ROMAN, Font.BOLD, 15));
-			jmi2.setFont(new Font(TIMES_NEW_ROMAN, Font.BOLD, 15));
-			jmi3.setFont(new Font(TIMES_NEW_ROMAN, Font.BOLD, 15));
-			jmi4.setFont(new Font(TIMES_NEW_ROMAN, Font.BOLD, 15));
-			jmi5.setFont(new Font("Times New Roman", Font.BOLD, 15));
-
-			jm1.add(jmi1);
-			jm1.add(jmi2);
-			jm2.add(jmi3);
-			jm2.add(jmi4);
-			jm3.add(jmi5);
-			jm4.add(jmi6);
-			jm4.add(jmi7);
-			jm4.add(jmi8);
-			jm4.add(jmi9);
-			jm5.add(jmi10);
-			jm5.add(jmi11);
-
-			jmb.add(jm1);
-			jmb.add(jm2);
-
-			jmb.add(jm4);
-			jmb.add(jm5);
-			jmb.add(jm3);
-
-
-			jmi1.addActionListener(this);
-			jmi1.setActionCommand("NewGame");
-			jmi2.addActionListener(this);
-			jmi2.setActionCommand("Exit");
-			jmi3.addActionListener(this);
-			jmi3.setActionCommand("Stop");
-			jmi4.addActionListener(this);
-			jmi4.setActionCommand("Continue");
-			jmi5.addActionListener(this);
-			jmi5.setActionCommand("help");
-			jmi6.addActionListener(this);
-			jmi6.setActionCommand("level1");
-			jmi7.addActionListener(this);
-			jmi7.setActionCommand("level2");
-			jmi8.addActionListener(this);
-			jmi8.setActionCommand("level3");
-			jmi9.addActionListener(this);
-			jmi9.setActionCommand("level4");
-			jmi10.addActionListener(this);
-			jmi10.setActionCommand("Player2");
-			jmi11.addActionListener(this);
-			jmi11.setActionCommand("Join");
-
-			this.setMenuBar(jmb);
-			this.setVisible(true);
+		for (int i = 0; i < metalWall.size(); i++) {
+			MetalWall mw = metalWall.get(i);
+			mw.draw(graphics);
+		}
 
-			for (int i = 0; i < 10; i++) {
-				if (i < 4)
-					homeWall.add(new CommonWall(350, 580 - 21 * i, this));

```

Comment:
Numeriai ComonWall klasėse kai yra kuriami yra nurodomi be paaiškinimo, todėl skaitant kodą ji sunku suprasti, bei palaikyti. Vietoje to, būtų gerai public sukurti static final int kintamuosius, kurie būtų fixir juos pridėti vietoje skaičių